components/camel-nats/src/main/java/org/apache/camel/component/nats/NatsProducer.java

```diff
@@ -42,21 +47,42 @@ public NatsEndpoint getEndpoint() {
     }
 
     @Override
-    public void process(Exchange exchange) throws Exception {
+    public void process(Exchange exchange) {
         NatsConfiguration config = getEndpoint().getConfiguration();
         byte[] body = exchange.getIn().getBody(byte[].class);
         if (body == null) {
-            // fallback to use string
-            body = exchange.getIn().getMandatoryBody(String.class).getBytes();
+            try {
+                // fallback to use string
+                body = exchange.getIn().getMandatoryBody(String.class).getBytes();
+            } catch (InvalidPayloadException e) {
+                exchange.setException(e);
+            }
         }
 
-        LOG.debug("Publishing to topic: {}", config.getTopic());
+        if (exchange.getPattern().isOutCapable()) {
+            LOG.debug("Requesting to topic: {}", config.getTopic());
 
-        if (ObjectHelper.isNotEmpty(config.getReplySubject())) {
-            String replySubject = config.getReplySubject();
-            connection.publish(config.getTopic(), replySubject, body);
+            CompletableFuture<io.nats.client.Message> future = connection.request(config.getTopic(), body);
+            try {
+                io.nats.client.Message msg = future.get(config.getRequestCleanupInterval(), TimeUnit.MILLISECONDS);
```

Comment:
the future get will block the current thread which is not ideal

Reply:
Yeah, I wasn't sure how to make the timeout work without blocking the thread.

Reply:
I think I managed to fix this.

Reply:
Seems to work, added a timeout test which passes at least.